app/src/main/kotlin/compiler/intermediate_form/Addressing.kt

```diff
@@ -0,0 +1,19 @@
+package compiler.intermediate_form
+
+typealias MemoryAddress = ULong
+
+sealed class Addressing {
```

Comment:
I think the classes here should extend `Addressing` (same problem for `IntermediateFormTreeNode`).

Reply:
Sure, thanks for the catch